common/dynamicconfig/constants.go

```diff
@@ -2077,6 +2077,12 @@ the user has not specified an explicit RetryPolicy`,
 		retrypolicy.DefaultDefaultRetrySettings,
 		`DefaultWorkflowRetryPolicy represents the out-of-box retry policy for unset fields
 where the user has set an explicit RetryPolicy, but not specified all the fields`,
+	)
+	AllowReusePolicyRejectWithConflictPolicyTerminate = NewNamespaceTypedSetting(
+		"history.allowReusePolicyRejectWithConflictPolicyTerminate",
+		false,
+		`Allows to use WorkflowIdReusePolicy RejectDuplicate with WorkflowIdReusePolicy TerminateExisting.
+If false, an invalid argument error is returned.`,
```

Comment:
nit: Might be worth having a single flag for this and changing the behavior of RejectDuplicateFailedOnly.

I'd also consider adding a warning that this flag may go away in upcoming releases, I don't think we want to keep this forever.

```suggestion
	AllowReusePolicyRejectWithConflictPolicyTerminate = NewNamespaceTypedSetting(
		"history.allowReusePolicyRejectWithConflictPolicyTerminate",
		false,
		`Allows to use WorkflowIdReusePolicy RejectDuplicate with WorkflowIdReusePolicy TerminateExisting.
If false (the default), an invalid argument error is returned.`,
```